Fabric.Identity.API/scripts/Register.ps1

```diff
@@ -0,0 +1,570 @@
+        $body = $implicitBody + $body
+    }
+
+    return $body
+}
+
+function Invoke-RegisterApiResources($apiResources, $identityServiceUrl, $accessToken)
+{
+    Write-Host "Registering APIs..."
+    Write-Host ""
+    if($apiResources -eq $null){
+        Write-Host "    No api resources"
+    }
+    foreach($api in $apiResources){
+        Write-Host "    Registering $($api.name)"
+        $body = Get-ApiResourceFromConfig($api)
+        $isApiRegistered = Test-IsApiRegistered -identityServiceUrl $identityServiceUrl -apiName $api.name -accessToken $accessToken
+        
+        if($isApiRegistered){
+            Write-Host "    API is registered, updating"
+            #$apiSecret = Invoke-UpdateApiRegistration -identityServiceUrl $identityServiceUrl -body $body -apiName $api.name -accessToken $accessToken
+        }else{
+            Write-Host "    API is not registered, adding"
+            $apiSecret = Add-ApiRegistration -authUrl $identityServiceUrl -body (ConvertTo-Json $body) -accessToken $accessToken
+        }
+    }
+}
+
+function Invoke-RegisterClients($clients, $identityServiceUrl, $accessToken, $discoveryServiceUrl)
+{
+    Write-Host "Registering Clients..."
+    Write-Host ""
+    if($clients -eq $null){
+        Write-Host "    No clients to register"
+    }
+    foreach($client in $clients){
+        Write-Host "    Registering $($client.clientid) with Fabric.Identity"
+        $body = Get-ClientFromConfig -client $client -discoveryServiceUrl $discoveryServiceUrl
+        $isClientRegistered = Test-IsClientRegistered -identityServiceUrl $identityServiceUrl -clientId $client.clientid -accessToken $accessToken
+
+        if($isClientRegistered){
+            Write-Host "    Client is registered, updating"
+            $clientSecret = Invoke-UpdateClientRegistration -identityServiceUrl $identityServiceUrl -body $body -clientId $client.clientid -accessToken $accessToken
+        }else{
+            Write-Host "    Client is not registered, adding"
+            $jsonBody = ConvertTo-Json $body
+            $clientSecret = Add-ClientRegistration -authUrl $identityServiceUrl -body $jsonBody -accessToken $accessToken 
+        }
+
+        Write-Host "    Registering $($client.clientid) with Fabric.Authorization"
+        $authorizationClient = Add-AuthorizationRegistration -authUrl $authorizationServiceURL -clientId $client.clientid -clientName $client.clientName -accessToken $accessToken
+
+        if($client.authorization -ne $null){
+            Invoke-RegisterRolesAndPermissions -grainName "app" -securableItemName $client.clientid -securableItem $client.authorization -identityServiceUrl $identityServiceUrl -accessToken $accessToken
+        }
+    }
+}
+
+function Invoke-RegisterSharedRolesAndPermissions($rolesAndPermissions, $identityServiceUrl, $accessToken){
+    Write-Host "Creating Roles and Permissions..."
+    Write-Host ""
```

Comment:
Has `$authorization` been declared/initialized within this function?

Reply:
Good catch, I changed the name of the variable in the function params, but didn't change it in this check.